src/IRrecv.cpp

```diff
@@ -21,6 +21,12 @@ extern "C" {
 #include "IRremoteESP8266.h"
 #include "IRutils.h"
 
+#if defined(ESP32)
+#if ( defined(ESP_ARDUINO_VERSION_MAJOR) && (ESP_ARDUINO_VERSION_MAJOR >= 3) )
+#include <driver/gpio.h>
```

Comment:
Does this hurt or cause an issue if it is always included?
i.e. Can we just include this header all the time?

Reply:
I'm more inclined to _only_ add includes when really needed, trying to avoid unneeded code being pulled in the build or having unexpected interactions.